### PR TITLE
Simulator task free mode

### PR DESCRIPTION
**Pull request by nagkumar91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_simulator.py [hunk 1]
```diff
@@ -586,7 +586,10 @@ async def _create_conversations_from_query_responses(
         for i, query_response_pair in enumerate(query_responses):
             query = query_response_pair["q"]
             response = query_response_pair["r"]
-            task = tasks[i]
+            try:
+                task = tasks[i]
```
Comment (luigiw) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_simulator.py:590]: When does this happen? Some query responses don't have a task associated with them?

File: sdk/evaluation/azure-ai-evaluation/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,12 @@
 ### Breaking Changes
 
 ### Bugs Fixed
+<<<<<<< HEAD
```
Comment (luigiw) [inline sdk/evaluation/azure-ai-evaluation/CHANGELOG.md:10]: Fix these.

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_simulator.py (hunk 1 above)
Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_simulator.py:590]: Yeah, there are occasions when users want 5 conversations, but only 3 tasks. Earlier, we would only return 3 conversations.

File: sdk/evaluation/azure-ai-evaluation/CHANGELOG.md (hunk 2 above)
Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/CHANGELOG.md]: resolved.

**Review by diondrapeck — APPROVED** [review]

**Pull request merged**